default.nix

```diff
@@ -1,22 +1,25 @@
-{ lib
+{ nix_lib
 , std
 , filter
 }: rec {
   common = rec {
-    toProtocInclude = x: lib.lists.fold (a: b: "-I=" + toString (a.src) + " " + b) "" x;
-    recursiveDeps = deps: proto_lib.lists.concatMap (y: (if builtins.length y.protoDeps != 0 then [ (recursiveDeps y.protoDeps) ] else [ ]) ++ [ y ]) deps;
-    recursiveProtoDeps = deps: proto_lib.lists.unique ((proto_lib.lists.flatten (recursiveDeps deps)));
+    inherit (lib.strings) splitString concatStringsSep;
+    inherit (lib.lists) fold flatten concatMap unique;
+    toProtocInclude = x: fold (a: b: "-I=" + toString (a.src) + " " + b) "" x;
+    recursiveDeps = deps: concatMap (y: (if builtins.length y.protoDeps != 0 then [ (recursiveDeps y.protoDeps) ] else [ ]) ++ [ y ]) deps;
+    recursiveProtoDeps = deps: unique ((flatten (recursiveDeps deps)));
+    slashToUnderscore = namespace: concatStringsSep "_" (splitString "/" namespace);
   };
 
-  proto_lib = std // lib // common;
+  lib = std // nix_lib // common;
 
-  generateMeta = { name, dir, version, protoDeps }:
+  generateMeta = { name, dir, version, protoDeps, namespace ? "" }:
```

Comment:
Could use namespace only for naming and only use "name" if namespace is not provided. Namespace is essentially the proto package and path so it should stand alone as the generated package name without "name".